### DISCUSSION TITLE
How to add a hook to lifecycle after view!{} render out?

### DISCUSSION THREAD

**Discussion by tempbottle** [leptos-rs/leptos#1561] [Q&A] [answered]
How to add a hook to lifecycle after view!{} render out in leptos?

code like this:
```
#[component]
pub fn SomeComp() -> impl IntoView {
    // Help: how to make ??? code run after view render out?
    ???
    view! {
        <div id="div_el_id"></div>
    }
}
```

**Comment by Miuler** [comment]
For what purpose do you want to do this?

**Reply by tempbottle** [reply]
There is an example:

https://github.com/yuankunzhang/charming/blob/main/examples/leptos-demo/src/main.rs#L10-L32

I want to display the chart without click when nav to the page.

and @gbj make a good answer, thanks.

**Comment by gbj** [comment, marked as answer]
Apologies, I missed this when you first posted it. I am assuming you mean that you want to run something on basically the tick after the component is rendered:

```rust
#[component]
pub fn SomeComp() -> impl IntoView {
    create_effect(|_| {
        request_animation_frame(|| {
            log!("view has rendered");
        });
    });
    view! {
        <div id="div_el_id"></div>
    }
}
```

Or do you mean something else by "after view render out"?

**Reply by tempbottle** [reply]
It works! thanks!
